src/hydrusvideodeduplicator/dedup.py

```diff
@@ -295,10 +322,14 @@ def _find_potential_duplicates(
 
                                 # Store last furthest searched position in the database for each element
                                 # This way you only have to start searching at that place instead of at i+1 if it exists
-                                row.setdefault("farthest_search_index", i + 1)
-
-                                # This is not necessary but may increase speed by avoiding any of the code below
-                                if row["farthest_search_index"] >= len(hashdb) - 1:
+                                farthest_search_index = row.setdefault("farthest_search_index", i + 1)
+
+                                if farthest_search_index > len(hashdb):
+                                    raise Exception(f"Found video at index {i} that had impossible " +
```

Comment:
@ianwal With the new `update_search_cache()` function being called before this, I'm pretty sure that this _should_ never happen. But if it ever does, we'd want to know about it.